modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItEmbeddedMarshallingTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.runner.app.client;
+
+import static org.apache.ignite.catalog.definitions.ColumnDefinition.column;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.ignite.catalog.ColumnType;
+import org.apache.ignite.catalog.definitions.TableDefinition;
+import org.apache.ignite.compute.JobDescriptor;
+import org.apache.ignite.compute.JobTarget;
+import org.apache.ignite.internal.runner.app.client.Jobs.ArgmarshallingJob;
+import org.apache.ignite.internal.runner.app.client.Jobs.ArgumentAndResultmarshallingJob;
+import org.apache.ignite.internal.runner.app.client.Jobs.ArgumentStringMarshaller;
+import org.apache.ignite.internal.runner.app.client.Jobs.JsonMarshaller;
+import org.apache.ignite.internal.runner.app.client.Jobs.PojoArg;
+import org.apache.ignite.internal.runner.app.client.Jobs.PojoJob;
+import org.apache.ignite.internal.runner.app.client.Jobs.PojoResult;
+import org.apache.ignite.internal.runner.app.client.Jobs.ResultStringUnMarshaller;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.table.Tuple;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for embedded API marshallers for Compute API.
+ */
+@SuppressWarnings("resource")
+public class ItEmbeddedMarshallingTest extends ItAbstractThinClientTest {
```

Comment:
We already have `ItThinClientMarshallingEmbeddedTest` (derived from `ItThinClientMarshallingTest`), do we need this class? I'd prefer to follow the pattern with a pair of client/embedded classes that test the same thing and ensure consistent behavior across both API implementations.

Reply:
I understand what you mean but I would not like to follow this pattern in Compute API. I think that the inheritance for tests in this case might introduce pitfalls in the future. I can see us having different contracts for thin/embedded clients when it comes to Compute. For example, deployment unit might not be used at all in the embedded mode. 

If you want me to extend some test cases and double-check the contracts I can do that.

Reply:
Then I don't understand why this test class is in the `client` package?

Reply:
makes sense, will move this test into `compute` package.